### PR TITLE
CarPort: Avalon 2021 TSS-P

### PR DESCRIPTION
**Pull request by rav4kumar**
- [x] added to README
- [x] test route added to [test_car_models](../../selfdrive/test/test_car_models.py)
- [x] route with openpilot: 399f98d4e88eb662|2021-01-07--18-58-27
- [x] route with stock system: 399f98d4e88eb662|2021-01-07--00-57-08 

@Glenn#6835 he has tested and verified it.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	url = ../../commaai/panda.git
 [submodule "opendbc"]
 	path = opendbc
-	url = ../../commaai/opendbc.git
+	url = ../../rav4kumar/opendbc.git
```
Comment (adeebshihadeh) [inline .gitmodules:6]: revert changes to this file

**Comment by adeebshihadeh** [comment]
Why does this need a new port? The current Avalon is also TSS-P.

File: .gitmodules (hunk 1 above)
Comment (rav4kumar) [reply .gitmodules:6]: done

**Comment by rav4kumar** [comment]
one of the signal is different and needs new dbc

**Comment by adeebshihadeh** [comment]
Which signal is different? If this is true, we'd also need an opendbc PR.

**Comment by rav4kumar** [comment]
BO_ 610 EPS_STATUS: 8 EPS
 SG_ IPAS_STATE : 3|4@0+ (1,0) [0|15] "" XXX
 SG_ LKA_STATE : 31|7@0+ (1,0) [0|127] "" XXX
 SG_ TYPE : 24|1@0+ (1,0) [0|1] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX

**Comment by adeebshihadeh** [comment]
Hmm, I just checked a route from our Rav 4, and the DBC is also wrong. I suspect many or all the Toyotas have the wrong length. Let's get this model added to the current Avalon instead of making a new one, and I'll fix the length for all the Toyotas in another PR.

**Comment by rav4kumar** [comment]
sounds good! thanks for checking. ill clean this up and let you know,

File: README.md [hunk 2]
```diff
@@ -98,6 +98,7 @@ Supported Cars
 | Lexus     | RX Hybrid 2016-19             | All               | Stock<sup>3</sup>| 0mph               | 0mph              |
 | Lexus     | RX Hybrid 2020                | All               | openpilot        | 0mph               | 0mph              |
 | Toyota    | Avalon 2016-18                | TSS-P             | Stock<sup>3</sup>| 20mph<sup>1</sup>  | 0mph              |
+| Toyota    | Avalon 2021                   | TSS-P             | Stock<sup>3</sup>| 20mph<sup>1</sup>  | 0mph              |
```
Comment (adeebshihadeh) [inline README.md:101]:
```suggestion
| Toyota    | Avalon 2016-18, 2021      | TSS-P             | Stock<sup>3</sup>| 20mph<sup>1</sup>  | 0mph              |
```

File: selfdrive/car/toyota/interface.py [hunk 3]
```diff
@@ -160,7 +160,7 @@ def get_params(candidate, fingerprint=gen_empty_fingerprint(), car_fw=[]):  # py
       ret.lateralTuning.pid.kpV, ret.lateralTuning.pid.kiV = [[0.18], [0.015]]  # community tuning
       ret.lateralTuning.pid.kf = 0.00012  # community tuning
 
-    elif candidate == CAR.AVALON:
+    elif candidate == [CAR.AVALON]:
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/interface.py:163]: revert changes to this file

File: selfdrive/car/toyota/values.py [hunk 4]
```diff
@@ -219,6 +219,10 @@ class CAR:
...
+  },
+  # Avalon 2021
+  {
+    36: 8, 37: 8, 114: 5, 170: 8, 180: 8, 186: 4, 355: 5, 426: 6, 452: 8, 464: 8, 466: 8, 467: 8, 544: 4, 550: 8, 552: 4, 562: 6, 608: 8, 610: 8, 643: 7, 705: 8, 728: 8, 740: 5, 761: 8, 764: 8, 767: 4, 800: 8, 810: 2, 812: 8, 822: 8, 824: 8, 829: 2, 830: 7, 835: 8, 836: 8, 869: 7, 870: 7, 871: 2, 888: 8, 891: 8, 898: 8, 900: 6, 902: 6, 905: 8, 913: 8, 918: 8, 921: 8, 933: 8, 934: 8, 935: 8, 944: 8, 945: 8, 951: 8, 955: 8, 956: 8, 976: 1, 998: 5, 999: 7, 1000: 8, 1001: 8, 1002: 8, 1014: 8, 1017: 8, 1020: 8, 1041: 8, 1042: 8, 1044: 8, 1056: 8, 1059: 1, 1076: 8, 1077: 8, 1112: 8, 1114: 8, 1161: 8, 1162: 8, 1163: 8, 1164: 8, 1165: 8, 1166: 8, 1167: 8, 1227: 8, 1235: 8, 1237: 8, 1263: 8, 1279: 8, 1412: 8, 1541: 8, 1552: 8, 1553: 8, 1556: 8, 1557: 8, 1568: 8, 1570: 8, 1571: 8, 1572: 8, 1595: 8, 1745: 8, 1779: 8, 1786: 8, 1787: 8, 1788: 8, 1789: 8, 1904: 8, 1912: 8, 1990: 8, 1998: 8
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/values.py:225]: remove this, FW versions should be sufficient

File: selfdrive/test/test_car_models.py [hunk 5]
```diff
@@ -250,6 +250,11 @@ def get_route_log(route_name):
     'enableCamera': False,
     'enableDsu': False,
   },
+  "399f98d4e88eb662|2021-01-07--18-58-27": {
+    'carFingerprint': TOYOTA.AVALON,
+    'enableCamera': True,
+    'enableDsu': False,
+  },
```
Comment (adeebshihadeh) [inline selfdrive/test/test_car_models.py:257]: no need for this since it's not a new model

**Comment by adeebshihadeh** [comment]
Also, revert the changes to opendbc

File: selfdrive/car/toyota/values.py [hunk 6]
```diff
@@ -298,23 +298,39 @@ class CAR:
 
 FW_VERSIONS = {
   CAR.AVALON: {
-    (Ecu.esp, 0x7b0, None): [b'F152607060\x00\x00\x00\x00\x00\x00'],
+    (Ecu.esp, 0x7b0, None): [
+      b'F152607060\x00\x00\x00\x00\x00\x00',
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/values.py:302]: why are there multiple fw versions added here?

Comment (rav4kumar) [reply selfdrive/car/toyota/values.py:302]: without those two it goes into dashcam

Comment (adeebshihadeh) [reply selfdrive/car/toyota/values.py:302]: I checked three routes and am consistently seeing `F152607180\x00\x00\x00\x00\x00\x00`. remove the other one

Comment (adeebshihadeh) [reply selfdrive/car/toyota/values.py:302]: if you'd like to add more FW, do it in another PR

File: selfdrive/car/toyota/values.py [hunk 7]
```diff
@@ -298,23 +298,39 @@ class CAR:
...
+      b'F152607110\x00\x00\x00\x00\x00\x00',
+      b'F152607180\x00\x00\x00\x00\x00\x00',
+    ],
     (Ecu.dsu, 0x791, None): [
       b'881510705200\x00\x00\x00\x00',
       b'881510701300\x00\x00\x00\x00',
+      b'881514106100\x00\x00\x00\x00',
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/values.py:309]: remove this one, same as above

File: selfdrive/car/toyota/values.py [hunk 8]
```diff
@@ -298,23 +298,39 @@ class CAR:
...
+      b'881510703200\x00\x00\x00\x00',
+    ],
+    (Ecu.eps, 0x7a1, None): [
+      b'8965B41051\x00\x00\x00\x00\x00\x00',
+      b'8965B41070\x00\x00\x00\x00\x00\x00',
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/values.py:314]: remove this one, same as above

File: selfdrive/car/toyota/values.py [hunk 9]
```diff
@@ -298,23 +298,39 @@ class CAR:
...
+      b'8965B41090\x00\x00\x00\x00\x00\x00',
     ],
-    (Ecu.eps, 0x7a1, None): [b'8965B41051\x00\x00\x00\x00\x00\x00'],
     (Ecu.engine, 0x7e0, None): [
       b'\x0230721100\x00\x00\x00\x00\x00\x00\x00\x00A0C01000\x00\x00\x00\x00\x00\x00\x00\x00',
       b'\x0230721200\x00\x00\x00\x00\x00\x00\x00\x00A0C01000\x00\x00\x00\x00\x00\x00\x00\x00',
     ],
+    (Ecu.engine, 0x700, None): [
+      b'\x01896630738000\x00\x00\x00\x00',
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/values.py:319]: remove this one, and avalon needs to be added here https://github.com/commaai/openpilot/blob/master/selfdrive/car/fw_versions.py#L127

**Comment by rav4kumar** [comment]
do you want me to make pr for dbc?

**Comment by adeebshihadeh** [comment]
No, I want to check all the Toyotas and fix them all at once. I already made an issue for this and would like to get to it before 0.8.2 release.

File: selfdrive/car/toyota/values.py (hunk 9 above)
Comment (adeebshihadeh) [reply selfdrive/car/toyota/values.py:319]: still needs to be removed

**Comment by adeebshihadeh** [comment]
All the submodules are changed. Once the submodule changes are reverted and the last extra FW version is removed, this should be good to merge.

**Comment by rav4kumar** [comment]
okay, sounds good! let me know if you need anything else.

File: selfdrive/car/toyota/values.py [hunk 10]
```diff
@@ -298,23 +298,34 @@ class CAR:
...
+    (Ecu.eps, 0x7a1, None): [
+      b'8965B41090\x00\x00\x00\x00\x00\x00',
     ],
-    (Ecu.eps, 0x7a1, None): [b'8965B41051\x00\x00\x00\x00\x00\x00'],
     (Ecu.engine, 0x7e0, None): [
       b'\x0230721100\x00\x00\x00\x00\x00\x00\x00\x00A0C01000\x00\x00\x00\x00\x00\x00\x00\x00',
       b'\x0230721200\x00\x00\x00\x00\x00\x00\x00\x00A0C01000\x00\x00\x00\x00\x00\x00\x00\x00',
     ],
+    (Ecu.engine, 0x700, None): [
+      b'\x01896630731000\x00\x00\x00\x00',
```
Comment (adeebshihadeh) [inline selfdrive/car/toyota/values.py:318]: this is wrong, should be
```suggestion
      b'\x01896630738000\x00\x00\x00\x00',
```

**Comment by adeebshihadeh** [comment]
fixed up in #19790

**Pull request closed without merging**